code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	..()
+	T = get_turf(src)
+	RegisterSignal(T, COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_rune)
+	cooldown = world.time + max_cooldown //rune starts on cooldown
+	
+/obj/effect/warped_rune/proc/clean_rune()
+	qdel(src)
+
+ ///using the extract on the floor will "draw" the rune.
+/obj/item/slimecross/warping/afterattack(turf/target, mob/user, proximity)
+	if(!proximity)
+		return
+
+	if(istype(target) && locate(/obj/effect/warped_rune) in target) //check if the target is a floor and if there's a rune on said floor
+		to_chat(user, "<span class='warning'>There is already a bluespace rune here!</span>")
+		return
+	if(istype(target, runepath))       //checks if the target is a rune and then if you can store it
+		if(warp_charge == max_charge)
+			to_chat(user, "<span class='warning'>Your crossbreed is already full!</span>")
+			return
+		if(do_after(user, storing_time,target = target) && warp_charge < max_charge)
+			to_chat(user, "<span class='notice'>You store the rune in your bluespace crossbreed</span>")
+			qdel(target)
+			warp_charge++
+			return
+
+	if(!istype(target) || isspaceturf(target))
+		return
+
+	if(locate(/turf/closed) in target || locate(/obj/structure) in target) // check if there's a wall or a structure in the way
+		to_chat(user, "<span class='warning'>something is in the way of the rune!</span>")
+		return
+
+	if(warp_charge > 0 && do_after(user, drawing_time,target = target)) //finally spawns the right rune if you have charge(s) left
+		playsound(target, 'sound/effects/slosh.ogg', 20, TRUE)
+
+		if(warp_charge == 0) //this is only here to fix a bug where the user can draw multiple runes at once if they spam click
+			to_chat(user, "<span class='warning'>Your crossbreed is empty!</span>")
+			return
+		warp_charge--
+		new runepath(target)
+		to_chat(user, "<span class='notice'>You carefully draw the rune with your crossbreed.</span>")
+
+
+/obj/item/slimecross/warping/grey
+	colour = "grey"
+	effect_desc = "Creates a rune.Extracts the rune is on are absorbed, 8 extracts produces a slime of that color."
+	runepath = /obj/effect/warped_rune/greyspace
+
+
+
+/obj/effect/warped_rune/greyspace
+	desc = "Death is merely a setback, anything can be rebuilt given the right components"
+	icon_state = "greyspace_rune"
+	var/i = 0 //number of slime extract currently absorbed by the rune
+	var/mob/living/simple_animal/slime/S //S stands for the slime that will be spawned
+	var/obj/item/slime_extract/extractype = 0 //extractype is used to remember the type of the extract on the rune
+	max_cooldown = 50 
+
+
+/obj/effect/warped_rune/greyspace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/greyspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		slimerevival()
+	
+///Makes a slime of the color of extract that were put on the runes.can only take one type of extract between slime spawning.
+/obj/effect/warped_rune/greyspace/proc/slimerevival()
```

Comment:
this rune is kind of stupidly inefficient. when would I ever use this?

IMO, you should make this rune instead convert ONE slime core and like 8 monkey cubes into a baby slime of that slime core's color. If the slime core was spent, the rune recharges it (to 1 charge) instead of making a baby slime from it. _That_ would be a rune that would at least see _some_ use. Remember, people are spending 11 slime cores to make each of these runes.